develop/learning-paths/migration/articles/01-intro/01-from-a-monolith-to-a-modular-architecture.markdown

```diff
@@ -0,0 +1,124 @@
+# From a Monolith to a Modular Architecture
+
+Since Liferay Portal's beginning it's been an amazing application full of all
+kinds of capabilities. It's had powerful built-in apps and extensible frameworks
+that deploy on all kinds of devices. These apps leverage Portal's powerful
+frameworks (e.g., asset and comment frameworks) and in some cases the apps
+leverage each other (e.g., the Blogs and Wiki apps leverage the Documents and
+Media Library). And undoubtedly you've leveraged Liferay Portal's apps and
+frameworks via their APIs. Yes, Liferay Portal has traditionally provided so
+many things to so many users and developers--and that's been a very good thing.
+
+But as Liferay Portal's grown, it has become a monolith of sorts. Its repository
+code base was huge and it's memory footprint was much bigger than what we'd
+prefer. Considering all that Portal had been providing and how it's evolved, its
+size isn't surprising, but we knew we had to do better for Liferay 7. So before
+we started designing Liferay 7, we took a step back to consider how to provide
+even better capabilities while making Liferay Portal, and its features and
+frameworks, lean, agile, and independent. To start, we took a hard look at key
+capabilities to improve.
+
+Liferay Portal's releasability was something we definitely wanted to improve.
+Before Liferay 7, you couldn't download releases of new features to *parts* of
+Liferay Portal--you had to wait until a new release of the *entire* portal
+product. And you couldn't just take the new parts you wanted, you had to take
+all of Liferay Portal--the whole monolith. Releasability of Liferay Portal's
+core and its components became a top priority.
+
+To improve releasability, we came face to face with how Portal's parts were
+interacting. In many cases, they had too much knowledge of each other's
+internals (i.e., they were coupled tightly). Tight coupling was occurring way too
+easily in the development process, as we hadn't yet found an easy way to facilitate and enforce
+loose coupling. So we also prioritized finding a framework to help Liferay
+engineers and contributors develop loosely coupled components on Liferay.
+
```

Comment:
Same thing. We don't want to say we had problems with our code, or that we did things in a non-optimal way. We just want to say we wanted to make improvements that both Liferay and the user/developer can benefit from.